src/pages/CookieClicker.vue

```diff
@@ -2,48 +2,186 @@
+    shinyCookie.value.visible = false;
+    spawnNextShiny(); // Schedule next shiny spawn
+}
+
+// Function to schedule the next shiny cookie spawn
+function spawnNextShiny() {
+    const delay = Math.random() * 20000 + 10000; // Random delay between 10-30 seconds
+    setTimeout(spawnShinyCookie, delay);
+}
+
+// Start first shiny cookie spawn
+spawnNextShiny();
 </script>
+
 <template>
     <div class="columns">
+        <!-- Main cookie counter -->
         <div class="column is-4 has-background-primary has-text-centered">
-            <h1 class="is-size-1">{{ +cookies.toFixed(1) }} cookies</h1>
-            <h3 class="is-size-3">{{ +cps.toFixed(1) }} cps</h3>
+            <h1 class="is-size-1">{{ cookies.toFixed(1) }} cookies</h1>
+            <h3 class="is-size-3">{{ cps.toFixed(1) }} cps</h3>
             <figure class="image is-square" @click="cookies++">
                 <img src="https://sweetlorens.com/cdn/shop/products/Copy-of-Chocolate-Chunk-Full-Cookie-transparent-background.png?v=1687811511" />
             </figure>
         </div>
-        <div class="column is-6 has-background-link">
-            asdas
+
+        <!-- Upgrades and Achievements section -->
+        <div class="column is-4 has-background-info">
+            <h2 class="is-size-2">Upgrades</h2>
+            <button v-for="upgrade in upgrades" 
+                    :disabled="cookies < upgrade.price || upgrade.bought" 
+                    @click="buyUpgrade(upgrade)" 
+                    class="button is-success is-fullwidth">
+                {{ upgrade.name }} 🍪{{ upgrade.price }} 
+                <span v-if="upgrade.bought">✅</span>
+            </button>
+
+            <h2 class="is-size-2">Achievements</h2>
+            <ul>
+                <li v-for="achievement in achieved" class="has-text-white">
+                    🎉 {{ achievement }}
+                </li>
+            </ul>
         </div>
-        <div class="column is-2 has-background-warning">
-            <button v-for="building in buildings" :disabled="cookies<building.price" @click="buyBuilding(building)" class="button is-primary is-large is-fullwidth">
+
+        <!-- Buildings section -->
+        <div class="column is-4 has-background-warning">
+            <h2 class="is-size-2">Buildings</h2>
+            <button v-for="building in buildings" 
+                    :disabled="cookies < building.price" 
+                    @click="buyBuilding(building)" 
+                    class="button is-primary is-fullwidth">
                 {{ building.name }} 🍪{{ building.price }} #{{ building.count }}
             </button>
         </div>
     </div>
-</template>
\ No newline at end of file
+
+    <!-- Shiny Cookie -->
+    <div v-if="shinyCookie.visible"
+         class="shiny-cookie"
+         :style="{ top: shinyCookie.y + 'vh', left: shinyCookie.x + 'vw' }"
+         @click="collectShinyCookie">
+        ✨🍪✨
+    </div>
+</template>
+
+<style>
+.columns {
+    margin: 20px;
```

Comment:
don't use pixels rem are better for responsivness and zooming etc. https://bulma.io/documentation/helpers/spacing-helpers/
theres no 20px but you could see maybe `m-3` or `m-2` are close